include/Graph/Graph.hpp

```diff
@@ -101,6 +101,8 @@ class Graph {
  private:
   T_EdgeSet<T> edgeSet = {};
 
+  std::shared_ptr<AdjacencyMatrix<T>> cachedAdjMatrix;
```

Comment:
Please use the `shared` alias for shared pointers (just for coherence with the rest of the code)

Reply:
Done. Please review.

Reply:
Also lines 253 ad 1472 please.

Reply:
Done. Please review.